docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
 # Developer Guide
 
+- [Design & implementation](#design--implementation)
+    - [Architecture](#architecture)  
+    - [Ui component](#ui-component)  
+    - [Logic component](#logic-component)  
+    - [Model component](#model-component)  
+    - [Storage component](#storage-component)
+- [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+- [User stories](#user-stories)
+- [Use cases](#use-cases)
+- [Non-functional requirements](#non-functional-requirements) 
+- [Glossary](#glossary)
+- [Instruction for manual testing](#instructions-for-manual-testing)
+
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+
+
+![Architecture](diagrams/Architecture.png)
+
+
+The Architecture Diagram above gives an overview of how different components interacts 
+with each other in the app. 
+
+As the app starts, the main program initialises the Ui, Logic and Storage components.
+Then, the Storage component reads existing data file if it is available, else a fresh copy of 
+data file is created. This step also initialises the content of runtime storage which is store 
+in a TaskMap object.
+
+Now the Ui component is ready to take inputs from the User, and the input is processed and executed
+in the Logic component, the execution result is passed back to Ui component and displayed to the User.
+Just before the main program exits, Storage component saves the content in the TaskMap to the data file.
+
+Commons provide utility functions and messages to be used by other components.
+
+The sequence diagram below shows a typical workflow the program.
+
+
+![ComponentsSeq](diagrams/ComponentsSeq.png)
```

Comment:
I am quite confused in the diagram because I cannot see when the classes are constructed. Maybe you can revise the diagram to show when the class construction also.